project/src/main/java/com/google/sps/data/PlacesFetcher.java

```diff
@@ -104,25 +107,40 @@ public PlacesFetcher(GeoApiContext geoApiContext) {
      *     for places or for places details
      */
     public ImmutableList<Place> fetch(UserPreferences preferences) throws FetcherException {
-        PlacesSearchResult[] placesSearchResult;
+        Map<String, Set<String>> placesSearchResults = new HashMap<>();
+        PlacesSearchResult[] resultsForCuisine;
+        // If user didn't choose any cuisines, search on all possible cuisines
+        Collection<String> cuisines =
+            (preferences.cuisines().isEmpty())
+                ? CUISINE_TO_SEARCH_WORDS.keySet() : preferences.cuisines();
         int attemptsCounter = 0;
         do {
-            attemptsCounter++;
-            try {
-                placesSearchResult = getPlacesSearchResults(
-                    genTextSearchRequest(preferences, INIT_SEARCH_RADIUS_M * attemptsCounter));
-            } catch (ApiException | InterruptedException | IOException | IllegalStateException e) {
-                throw new FetcherException("Couldn't fetch places from Places API", e);
+            for (String cuisine: cuisines) {
+                attemptsCounter++;
+                try {
+                    resultsForCuisine = getPlacesSearchResults(genTextSearchRequest(
+                        preferences, INIT_SEARCH_RADIUS_M * attemptsCounter, cuisine));
+                } catch (ApiException
+                        | InterruptedException
+                        | IOException
+                        | IllegalStateException e) {
+                    throw new FetcherException("Couldn't fetch places from Places API", e);
```

Comment:
Something to think about - if one cuisine fails that means we want to throw an error for the entire request? 
I think maybe if we get some results without one of the cuisines it's better than eliminating the whole request (We can also inform the user about which cuisines if we want, of course not necessarily in this PR)

Reply:
That's a really good point, actually it's relevant before this PR also because it could've failed in one iteratoin in the radius extensoions and nothing would be returned. I will add this as a TODO and try treating this in another PR.